sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala

```diff
@@ -268,7 +268,7 @@ case class StringSplit(str: Expression, pattern: Expression)
   usage = "_FUNC_(str, regexp, rep) - Replaces all substrings of `str` that match `regexp` with `rep`.",
   extended = """
     Examples:
-      > SELECT _FUNC_('100-200', '(\d+)', 'num');
+      > SELECT _FUNC_('100-200', '(\\d+)', 'num');
```

Comment:
Hmm, when I wrote the docs on line 160, I was suggested to use unescaped characters.

> Since Spark 2.0, string literals (including regex patterns) are unescaped in our SQL parser. For example, to match "\abc", a regular expression for `regexp` can be "^\\abc$".

Actually, you need to write like this in spark-shell:

    scala> sql("SELECT like('\\\\abc', '\\\\\\\\abc')").show
    +---------------+
    |\abc LIKE \\abc|
    +---------------+
    |           true|
    +---------------+

    scala> sql("SELECT regexp_replace('100-200', '(\\\\d+)', 'num')").show
    +-----------------------------------+
    |regexp_replace(100-200, (\d+), num)|
    +-----------------------------------+
    |                            num-num|
    +-----------------------------------+


The behavior of Spark 2 when parsing SQL string literal reads `\\\\abc`  as `\abc` and `(\\\\d+)` as `(\d+)` in spark-shell.

But in spark-sql, you write the queries like this:

    spark-sql> SELECT like('\\abc', '\\\\abc');
    true
    Time taken: 0.061 seconds, Fetched 1 row(s)

    spark-sql> SELECT regexp_replace('100-200', '(\\d+)', 'num');
    num-num
    Time taken: 0.117 seconds, Fetched 1 row(s)

So depending how the shell environment processes string escaping, the query looks different. In the docs, it seems to me that writing in unescaped style can avoid this confusion?